### ISSUE TITLE
BigFloat promotion failure

### ISSUE THREAD

**Issue by MikaelSlevinsky** [ApproxFun/ApproxFun.jl#197]
What is going on here?
```julia
julia> using ApproxFun

julia> d = Interval{BigFloat}()
【-1e+00,1e+00】

julia> [ldirichlet(d) 0]
1x2 Array{Operator{BigFloat},2}:
 Evaluation{Ultraspherical{0},Bool,BigFloat}(Chebyshev(【-1e+00,1e+00】),false,0)  ZeroOperator{BigFloat,AnySpace,ZeroSpace}(AnySpace(),ZeroSpace())

julia> [ldirichlet(d) Derivative(d) 0]
1x3 Array{Operator{BigFloat},2}:
 Evaluation{Ultraspherical{0},Bool,BigFloat}(Chebyshev(【-1e+00,1e+00】),false,0)  Derivative{Ultraspherical{0},Int64,BigFloat}(Chebyshev(【-1e+00,1e+00】),1)  ZeroOperator{BigFloat,AnySpace,ZeroSpace}(AnySpace(),ZeroSpace())

julia> [Derivative(d) 0]
ERROR: type: arrayset: expected BandedOperator{BigFloat}, got ZeroOperator{Float64,AnySpace,ZeroSpace}
 in setindex! at multidimensional.jl:68
 in cat at abstractarray.jl:664
 in hcat at abstractarray.jl:671
```

**Comment by MikaelSlevinsky** [comment]
Here's another:
```julia
julia> using ApproxFun

julia> d = Interval{BigFloat}()
【-1e+00,1e+00】

julia> x = Fun(identity,d)
Fun(BigFloat[0e+00,1e+00],Chebyshev(【-1e+00,1e+00】))

julia> D = Derivative(d)
Derivative{Ultraspherical{0},Int64,BigFloat}(Chebyshev(【-1e+00,1e+00】),1)

julia> (x*D^2)[1:10,1:10]
10x10 BandedMatrix{BigFloat}:
 0e+00  0e+00  0e+00  3.99999999999999977795539507e+00  0e+00  0e+00                             0e+00                             0e+00                             0e+00  0e+00                           
 0e+00  0e+00  1e+00  0e+00                             5e+00  0e+00                             0e+00                             0e+00                             0e+00  0e+00                           
 0e+00  0e+00  0e+00  1.99999999999999988897769754e+00  0e+00  5.99999999999999977795539507e+00  0e+00                             0e+00                             0e+00  0e+00                           
 0e+00  0e+00  0e+00  0e+00                             3e+00  0e+00                             7.00000000000000044408920985e+00  0e+00                             0e+00  0e+00                           
 0e+00  0e+00  0e+00  0e+00                             0e+00  4.00000000000000022204460493e+00  0e+00                             7.99999999999999955591079015e+00  0e+00  0e+00                           
 0e+00  0e+00  0e+00  0e+00                             0e+00  0e+00                             5.00000000000000022204460493e+00  0e+00                             9e+00  0e+00                           
 0e+00  0e+00  0e+00  0e+00                             0e+00  0e+00                             0e+00                             5.99999999999999966693309261e+00  0e+00  1.00000000000000004440892099e+01
 0e+00  0e+00  0e+00  0e+00                             0e+00  0e+00                             0e+00                             0e+00                             7e+00  0e+00                           
 0e+00  0e+00  0e+00  0e+00                             0e+00  0e+00                             0e+00                             0e+00                             0e+00  7.99999999999999955591079015e+00
 0e+00  0e+00  0e+00  0e+00                             0e+00  0e+00                             0e+00                             0e+00                             0e+00  0e+00                           
```

**Comment by MikaelSlevinsky** [comment]
The second is fixed @834b5217e164e4393b295ef60824ba8c459c107d.

**Comment by MikaelSlevinsky** [comment]
The first is fixed @16ac3ce3ce572106931b4f388bf8b7a925abaf55.

**Comment by MikaelSlevinsky** [comment]
I'm going to leave this open in case anyone else happens across promotion failures in BigFloat, though I won't post the ones I find & fix.
